### PR TITLE
Jordan Lemma type Contour Integral

### PR DESCRIPTION
**Pull request by Teut2711**
#### Release Notes

* complexanalysis
  * complex analysis - integrals, theorems, inverse Laplace transforms
* Contour Integral
  * Currently only Jordan's lemma type integrals are solvable.

### CODE REVIEW

**Comment by XtremeGood** [comment]
As suggested by @Arif Ahmed I have made the most basic thing. Please comment on what and how the development should go further so that the project can be made one such that it could be put for GSoC.

**Comment by XtremeGood** [comment]
I have created a new submodule. I don't understand why sympy bot is should me errors about that.

**Comment by supreet11agrawal** [comment]
> I have created a new submodule. I don't understand why sympy bot is showing me errors in that.

The new submodule is not added to the list in https://github.com/sympy/sympy-bot/blob/master/sympy_bot/submodules.txt
This is creating the problem here

**Comment by XtremeGood** [comment]
Ok, I will add it there. Can I get some review on the code (not the quality yet but the working) and get an agreement if this should be a good enough project for GSoC?

**Comment by XtremeGood** [comment]
Module would include:
Contour integration
Graphing in argand plane
Inverse Laplace transforms
at the least..

File: sympy/complexanalysis/Contour Integration.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            #Enter in anticlockwise direction
+            self.solveit = solveit_oth()
+            self.points = points
+
+    def contour(self):
+
+        #Plot the contour
+        pass   
+
+    def residue_calc(self, rooot, mult, x ):
```
Comment (asmeurer) [inline sympy/complexanalysis/Contour Integration.py]: There is already a `residue` function that can be used here.

Comment (XtremeGood) [reply sympy/complexanalysis/Contour Integration.py]: Ok, I will substitute it here. But should I work on it?

Comment (XtremeGood) [reply sympy/complexanalysis/Contour Integration.py]: Can it be a good enough project for GSoC?

**Comment by asmeurer** [comment]
I guess this is meant to be a proof of concept more than anything. 

My main concern with contour integration is, how do you ensure correctness. To do this, you need to first make sure you find all the poles of the function. If you miss a simple pole, you will get the wrong answer. `solveset` can give guarantees on finding every root, but it may need some improvement to handle circular domains in the complex plane. See also the `singularities` function.  The other issue is branch cuts, and other nonstandard singularities.

**Comment by XtremeGood** [comment]
Yes you are right. Also, no one can say how to select a contour. But my main objective is to solve mellin's integrals by contour integration.

**Comment by XtremeGood** [comment]
I would modify it to first "dress" the function, i.e if you see trigonometric substitution also requires conversion to e^ix and picking out the real or im part. Every module has some or the other trouble but we  start it and then patches over patches come up with a solution. Therefore I think I should go about it.

**Comment by XtremeGood** [comment]
Some integrals may be evaluated wrong agreed but there would be many solved correctly and this is what is also used in physics, isn't it?

**Comment by XtremeGood** [comment]
And ya poles of a function can be found by Laurent series expansion.

**Comment by czgdp1807** [comment]
It appears that the PR can be closed as far as I can figure out from the reviews. However if someone changes their mind and thinks of re-opening it. Please let me know.
Thanks for the contributions and effort. Keep contributing. Closing for now.

**Comment by XtremeGood** [comment]
I proposed in my last GSoC proposal to build the system which was rejected. I have experience using Sympy but I need some serious help to make it and 'm still expecting this year if someone accepts my proposal and help me build it.

**Comment by czgdp1807** [comment]
IMHO, it would be great to start a discussion on the google groups of SymPy on this topic. We can discuss algorithms, APIs and other constraints there. Thanks for proposing this idea.

**Comment by XtremeGood** [comment]
So do you want to proceed or not?

**Comment by XtremeGood** [comment]
I saw once that the implementation of Laplace technique to solve differential equations is already implemented in some doc.

**Pull request closed without merging**